query/src/storages/fuse/operations/read_partitions.rs

```diff
@@ -75,10 +75,13 @@ impl FuseTable {
         blocks_metas: &[BlockMeta],
         push_downs: Option<Extras>,
     ) -> (Statistics, Partitions) {
-        let limit = push_downs
-            .as_ref()
-            .and_then(|p| p.limit)
-            .unwrap_or(usize::MAX);
+        let mut limit = usize::MAX;
```

Comment:
How about:

```rust
let limit = push_downs
            .as_ref()
            .filter(|p| p.order_by.is_empty())
            .and_then(|p| p.limit)
            .unwrap_or(usize::MAX);
```

Reply:
> How about:
> 
> ```rust
> let limit = push_downs
>             .as_ref()
>             .filter(|p| p.order_by.is_empty())
>             .and_then(|p| p.limit)
>             .unwrap_or(usize::MAX);
> ```

That's great, I did not come up with this idea.